### PR TITLE
Incorrect snapshot deletion algorithm #26584

### PR DESCRIPTION
**Pull request by helena**
#26584 WIP 
* Test update on params per issue suggestion, and a few minor untyped test deprecation/warning fixes.

### CODE REVIEW

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala [hunk 1]
```diff
@@ -762,31 +762,30 @@ class EventSourcedBehaviorSpec extends ScalaTestWithActorTestKit(EventSourcedBeh
           ctx ⇒
             counterWithSnapshotAndRetentionProbe(ctx, pid, snapshotProbe.ref, retentionProbe.ref)
               .snapshotEvery(2)
-              .withRetention(
-                RetentionCriteria(snapshotEveryNEvents = 2, keepNSnapshots = 2, deleteEventsOnSnapshot = false))))
+              .withRetention(RetentionCriteria(snapshotEveryNEvents = 3, keepNSnapshots = 2))))
```
Comment (helena) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala]: @patriknw this still passes oddly?

Comment (helena) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala]: Still investigating.

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala [hunk 2]
```diff
@@ -762,31 +762,30 @@ class EventSourcedBehaviorSpec extends ScalaTestWithActorTestKit(EventSourcedBeh
...
       persistentActor ! GetValue(replyProbe.ref)
-      replyProbe.expectMessage(State(3, Vector(0, 1, 2)))
+      replyProbe.expectMessage(State(10, (0 until 10).toVector))
       snapshotProbe.expectMessage(Try(Done))
       retentionProbe.expectMessageType[Success[DeleteSnapshotsCompleted]].value match {
         case DeleteSnapshotsCompleted(Criteria(SnapshotSelectionCriteria(maxSequenceNr, _, minSequenceNr, _))) =>
-          maxSequenceNr shouldEqual 2
-          minSequenceNr shouldEqual 0
+          maxSequenceNr shouldEqual 9
+          minSequenceNr shouldEqual 3
```
Comment (helena) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala]:
```scala
.snapshotEvery(2)
.withRetention(RetentionCriteria(snapshotEveryNEvents = 3, keepNSnapshots = 2)
```
* Store 0-9
* Snapshot delete from 3 to 9
* Store 10-19
* Snapshot delete from 13 to 19

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala]: Note snapshotEvery(2) vs snapshotEveryNEvents = 3

They must be the same, we have another issue for that duplication in the api.

Comment (helena) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala]: Ok, that duplication is one thing I was unsure of. Given sharding has that usage of snapshotEvery in it's delete usage.

**Comment by helena** [comment]
Test does
```scala
.snapshotEvery(3)
.withRetention(RetentionCriteria(snapshotEveryNEvents = 3, keepNSnapshots = 2)
```
* Store 1 - 10
* S1(1,2,3) S2(4,5,6) S3(7,8,9)
* Snapshot delete from 5 to 9
* Store 10-19
* Snapshot delete from 15 to 19

**Comment by patriknw** [comment]
Superseded by https://github.com/akka/akka/pull/26595

**Pull request closed without merging**